Formula/mpich.rb

```diff
@@ -48,22 +59,33 @@ def install
       system "./autogen.sh"
     end
 
-    system "./configure", "--disable-dependency-tracking",
-                          "--enable-fast=all,O3",
-                          "--enable-g=dbg",
-                          "--enable-romio",
-                          "--enable-shared",
-                          "--with-pm=hydra",
-                          "FC=gfortran-#{Formula["gcc"].any_installed_version.major}",
-                          "F77=gfortran-#{Formula["gcc"].any_installed_version.major}",
-                          "--disable-silent-rules",
-                          "--prefix=#{prefix}",
-                          "--mandir=#{man}",
-                          # Flag for compatibility with GCC 10
-                          # https://lists.mpich.org/pipermail/discuss/2020-January/005863.html
-                          "FFLAGS=-fallow-argument-mismatch",
-                          "CXXFLAGS=-Wno-deprecated",
-                          "CFLAGS=-fgnu89-inline -Wno-deprecated"
+    args = %W[
+      --disable-dependency-tracking
+      --enable-fast=all,O3
+      --enable-g=dbg
+      --enable-romio
+      --enable-shared
+      --with-pm=hydra
+      FC=gfortran-#{Formula["gcc"].any_installed_version.major}
+      F77=gfortran-#{Formula["gcc"].any_installed_version.major}
+      --disable-silent-rules
+      --prefix=#{prefix}
+      --mandir=#{man}
+    ]
+
+    # Flag for compatibility with GCC 10
+    # https://lists.mpich.org/pipermail/discuss/2020-January/005863.html
+    args << "FFLAGS=-fallow-argument-mismatch"
+    args << "CXXFLAGS=-Wno-deprecated"
+    args << "CFLAGS=-fgnu89-inline -Wno-deprecated"
```

Comment:
@iMichka I am tracking down a downstream issue building a C++ CMake project using `FindMPI.cmake` against Brew's MPICH on macOS-10.15.

Since this update, it looks like this line's `-fgnu89-inline` for some reason ends on the downstream compiler C++ line ending in a:
```
[  3%] Building CXX object CMakeFiles/<project>.dir/src/<file>.cpp.o
  clang: warning: -framework -std=c++14: 'linker' input unused [-Wunused-command-line-argument]
  error: invalid argument '-fgnu89-inline' not allowed with 'C++'
```

I posted an issue & reproducer in #80465